### PR TITLE
Remove use of httptest.ResponseRecorder in header tests

### PR DESCRIPTION
**Pull request by mattiasgrenfeldt**
Addresses #27 for tests in `tests/integration/header/header_test.go`.

### CODE REVIEW

File: tests/integration/header/header_test.go [hunk 1]
```diff
@@ -42,6 +43,28 @@ func (dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, d
 	return nil
 }
 
+type responseWriter struct {
```
Comment (kele) [inline tests/integration/header/header_test.go:46]: nit: `fakeResponseWriter`

File: tests/integration/header/header_test.go [hunk 2]
```diff
@@ -42,6 +43,28 @@ func (dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, d
...
+	header http.Header
+	writer io.Writer
+	status int
+}
+
+func newResponseWriter(w io.Writer) *responseWriter {
+	return &responseWriter{header: http.Header{}, writer: w, status: http.StatusOK}
+}
+
+func (r responseWriter) Header() http.Header {
```
Comment (kele) [inline tests/integration/header/header_test.go:56]: Keep the receiver type (pointer or not) the same across all methods.

File: tests/integration/header/header_test.go (hunk 1 above)
Comment (kele) [reply tests/integration/header/header_test.go:46]: Just checked other files: feel free to disregard this for consistency. It's not terribly important.

File: tests/integration/header/header_test.go (hunk 2 above)
Comment (mattiasgrenfeldt) [reply tests/integration/header/header_test.go]: Fixed. But why do you want this? Having just a normal receiver sometimes signals that the function is read-only.

Comment (kele) [reply tests/integration/header/header_test.go]: Well, not exactly.

Value receivers are used for passing objects *as values* (copies). Having a value receiver doesn't prevent you from altering objects that `responseWriter` references (e.g. slices, maps, pointers). So it's at best a "shallow" read-only.

Comment (mattiasgrenfeldt) [reply tests/integration/header/header_test.go]: Right!

**Pull request merged**